drivers/shtc1/shtc1.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2017 RWTH-Aachen
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_shtc1
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the Sensirion SHTC1 temperature
+ *              and humidity sensor.
+ *
+ * @author      Steffen Robertz <steffen.robertz@rwth-aachen.de>
+ * @author      Josua Arndt <jarndt@ias.rwth-aachen.de>
+ * @}
+ */
+#include <string.h>
+#include <math.h>
+
+#include "log.h"
+#include "assert.h"
+#include "checksum/crc8.h"
+#include "shtc1.h"
+#include "shtc1_regs.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+int8_t shtc1_init(shtc1_t *const dev, const shtc1_params_t *params)
+{
+    uint16_t id;
+
+    /* check for a valid device descriptor and parameters */
+    assert(dev && params);
+    /* copy settings into the device descriptor */
+    dev->params = *params;
+    /* Verify the connection by reading the SHTC1's ID and checking its value */
+    if (shtc1_id(dev, &id) != SHTC1_OK) {
+        return SHTC1_ERROR_BUS;
+    }
+    else if ((id & 0x3F) != SHTC1_ID) {
+        return SHTC1_ERROR_CRC;
+    }
+    else {
+        return SHTC1_OK;
+    }
+}
+
+int8_t shtc1_read(const shtc1_t *dev, uint16_t *rel_humidity,
+                  int16_t *temperature)
+{
+    uint8_t received[6];
+    /* Build and issue the measurement command */
+    uint8_t cmd[] =
+    { SHTC1_MEASURE_CLOCK_STRETCHING_TEMP_HIGH,
+      SHTC1_MEASURE_CLOCK_STRETCHING_TEMP_LOW };
+
+    i2c_acquire(dev->params.i2c_dev);
+    if (i2c_write_bytes(dev->params.i2c_dev, dev->params.i2c_addr, cmd, 2, 0)) {
+
```

Comment:
I meant adding a new line after this if block and before the block comment below.